src/pages/NewTodoPage.tsx

```diff
@@ -1,7 +1,60 @@
-import React from "react";
+import React, { useContext } from "react";
+import InputDetail from "../components/InputDetail";
+import { Container } from "@mui/system";
+import GlobalContext from "../contexts/global";
+import { useNavigate, useParams } from "react-router-dom";
 
 const NewTodoPage = () => {
-  return <div>NewTodoPage</div>;
+  const { state, setState } = useContext(GlobalContext);
+  const {categories} = state;
+
+  const [taskTitle, setTaskTitle] = React.useState<string>("");
+  const [taskDescription, setTaskDescription] = React.useState<string>("");
+  const [taskFrequency, setTaskFrequency] = React.useState<string>("");
+
+  const navigate = useNavigate();
+
+  const {id} = useParams();
+
+  const handleAdd = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    // @ts-ignore
+    const category = categories[id];
+    
+    const newTask = {
+      title: taskTitle,
+      description: taskDescription,
+      frequency: taskFrequency,
+      id: Date.now(),
+    };
+    category.tasks = [...category.tasks, newTask];
+    
+    setState({
+      ...state,
+      categories: {
+        ...state.categories,
+        // @ts-ignore
+        [id]: {
+         ...category, 
+        }
+      },
+    });
```

Comment:
not totally beautiful here, but this is how i'm adding a task to a specific category.

first find the category on line 23
then create the `newTask` object with the values from the inputs being tracked by the various `useStates`
then set `category.tasks` equal to the existing tasks, plus the new `newTask` on line 31
then finally call setState, using the `id` to update the correct category object in `state.categories`